pbjson-types/src/any.rs

```diff
@@ -0,0 +1,248 @@
+            serde::ser::Error::custom(
+                "Couldn't transcode google.protobuf.Any value into google.protobuf.Value",
+            )
+        })?;
+
+        let mut field_length = 1;
+        match value.kind {
+            Some(crate::value::Kind::StructValue(map)) => {
+                field_length += map.len();
+                let mut map_ser = serializer.serialize_map(Some(field_length))?;
+                map_ser.serialize_entry(TYPE_FIELD, &self.type_url)?;
+
+                for (k, v) in &map {
+                    map_ser.serialize_entry(k, v)?;
+                }
+
+                map_ser.end()
+            }
+            _ => {
+                let mut struct_ser = serializer.serialize_struct(NAME, 2)?;
+                struct_ser.serialize_field(TYPE_FIELD, &self.type_url)?;
+                struct_ser.serialize_field(VALUE_FIELD, &base64::encode(&self.value))?;
+                struct_ser.end()
+            }
+        }
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for crate::Any {
+    #[allow(deprecated)]
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        deserializer.deserialize_map(AnyVisitor)
+    }
+}
+
+struct AnyVisitor;
+
+impl<'de> serde::de::Visitor<'de> for AnyVisitor {
+    type Value = crate::Any;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        formatter.write_str("struct google.protobuf.Any")
+    }
+
+    fn visit_map<V>(self, mut map_access: V) -> Result<Self::Value, V::Error>
+    where
+        V: serde::de::MapAccess<'de>,
+    {
+        enum BytesOrValue {
+            Bytes(bytes::Bytes),
+            Value(crate::Value),
+        }
+
+        let mut type_url = None;
+        let mut value: Option<BytesOrValue> = None;
+        let mut map = std::collections::HashMap::new();
+        while let Some(k) = dbg!(map_access.next_key())? {
+            match k {
+                AnyField::TypeUrl => {
+                    if type_url.is_some() {
+                        return Err(serde::de::Error::duplicate_field(TYPE_FIELD));
+                    }
+                    type_url = Some(map_access.next_value()?);
+                }
+                AnyField::Value => {
+                    if value.is_some() {
+                        return Err(serde::de::Error::duplicate_field(VALUE_FIELD));
+                    }
+
+                    value = if let Ok(bytes) =
+                        dbg!(map_access.next_value::<::pbjson::private::BytesDeserialize<_>>())
+                    {
+                        Some(BytesOrValue::Bytes(bytes.0))
+                    } else {
+                        Some(BytesOrValue::Value(map_access.next_value()?))
+                    };
+                }
+                AnyField::Unknown(key) => {
+                    if map.contains_key(&key) {
```

Comment:
Surely this should be an error?

Reply:
What do you mean? It does return an error?

Reply:
I meant any unknown field should be an error, regardless of if it already exists in the map?